eclass/dotnet-pkg-utils.eclass

```diff
@@ -0,0 +1,606 @@
+
+# @ECLASS_VARIABLE: DOTNET_OUTPUT
+# @DEFAULT_UNSET
+# @DESCRIPTION:
+# Path of the output directory, where the package artifacts are placed during
+# the building of packages with "dotnet-pkg-utils_build" function.
+#
+# This variable is set automatically by the "dotnet-pkg-utils_setup" function.
+
+# @VARIABLE: DOTNET_LAUNCHERDEST
+# @INTERNAL
+# @DESCRIPTION:
+# Sets the path that .NET launchers are installed into by
+# the "dotnet-pkg-utils_dolauncher" function.
+#
+# The function "dotnet-pkg-utils_launcherinto" is able to manipulate this
+# variable.
+#
+# Defaults to "/usr/bin".
+DOTNET_LAUNCHERDEST=/usr/bin
+
+# @VARIABLE: DOTNET_LAUNCHERVARS
+# @INTERNAL
+# @DESCRIPTION:
+# Sets additional variables for .NET launchers created by
+# the "dotnet-pkg-utils_dolauncher" function.
+#
+# The function "dotnet-pkg-utils_append_launchervar" is able to manipulate this
+# variable.
+#
+# Defaults to a empty array.
+DOTNET_LAUNCHERVARS=()
+
+# @FUNCTION: dotnet-pkg-utils_get-configuration
+# @DESCRIPTION:
+# Return .NET configuration type of the current package.
+#
+# It is advised to refer to the "DOTNET_CONFIGURATION" variable instead of
+# calling this function if necessary.
+#
+# Used by "dotnet-pkg-utils_setup".
+dotnet-pkg-utils_get-configuration() {
+	if in_iuse debug && use debug ; then
+		echo Debug
+	else
+		echo Release
+	fi
+}
+
+# @FUNCTION: dotnet-pkg-utils_get-output
+# @USAGE: <name>
+# @DESCRIPTION:
+# Return a specially constructed name of a directory for output of
+# "dotnet build" artifacts ("--output" flag, see "dotnet-pkg-utils_build").
+#
+# It is very rare that a maintainer would use this function in an ebuild.
+#
+# This function is used inside "dotnet-pkg-utils_setup".
+dotnet-pkg-utils_get-output() {
+	debug-print-function ${FUNCNAME} "${@}"
+
+	[[ ! ${DOTNET_CONFIGURATION} ]] &&
+		die "${FUNCNAME}: DOTNET_CONFIGURATION is not set."
+
+	echo "${WORKDIR}"/${1}_net${DOTNET_COMPAT}_${DOTNET_CONFIGURATION}
+}
+
+# @FUNCTION: dotnet-pkg-utils_get-runtime
+# @DESCRIPTION:
+# Return the .NET runtime used for the current package.
+#
+# Used by "dotnet-pkg-utils_setup".
+dotnet-pkg-utils_get-runtime() {
+	local libc="$(usex elibc_musl "-musl" "")"
+
+	if use amd64 ; then
+		echo linux${libc}-x64
+	elif use x86 ; then
+		echo linux${libc}-x86
+	elif use arm ; then
+		echo linux${libc}-arm
+	elif use arm64 ; then
+		echo linux${libc}-arm64
+	else
+		die "${FUNCNAME}: Unsupported architecture: ${ARCH}"
+	fi
+}
+
+# @FUNCTION: dotnet-pkg-utils_setup
+# @DESCRIPTION:
+# Sets up "DOTNET_EXECUTABLE" variable for later use in "edotnet".
+# Also sets up "DOTNET_CONFIGURATION" and "DOTNET_OUTPUT"
+# for "dotnet-pkg_src_configure" and "dotnet-pkg_src_compile".
+#
+# This functions should be called by "pkg_setup".
+#
+# Used by "dotnet-pkg_pkg_setup" from the "dotnet-pkg" eclass.
+dotnet-pkg-utils_setup() {
+	local dotnet_compat_impl
+	local dotnet_compat_impl_path
```

Comment:
Any specific reason you're using such a long variable names?

Reply:
Grep, rg, occur.